From 2e69a8f18da6895f8ec2d6f334d7fb72c23e4d91 Mon Sep 17 00:00:00 2001
From: Maarten Bosmans <mkbosmans@gmail.com>
Date: Sat, 19 Mar 2011 13:59:15 +0100
Subject: [PATCH 1/3] Get rid of some warnings: -Wunused-result
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

modules/module-default-device-restore.c: In function ‘load’:
modules/module-default-device-restore.c:67: warning: ignoring return value of ‘fgets’,
declared with attribute warn_unused_result [-Wunused-result]
modules/module-default-device-restore.c:88: warning: ignoring return value of ‘fgets’,
declared with attribute warn_unused_result [-Wunused-result]

pulsecore/authkey.c: In function ‘generate’:
pulsecore/authkey.c:58: warning: ignoring return value of ‘ftruncate’,
declared with attribute warn_unused_result [-Wunused-result]

pulsecore/core-util.c: In function ‘pa_make_secure_dir’:
pulsecore/core-util.c:261: warning: ignoring return value of ‘fchown’,
declared with attribute warn_unused_result [-Wunused-result]
---
 src/modules/module-default-device-restore.c | 8 ++++----
 src/pulsecore/authkey.c                     | 5 ++++-
 src/pulsecore/core-util.c                   | 7 ++++---
 3 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/src/modules/module-default-device-restore.c b/src/modules/module-default-device-restore.c
index 27ae60e..06bbd65 100644
--- a/src/modules/module-default-device-restore.c
+++ b/src/modules/module-default-device-restore.c
@@ -64,8 +64,8 @@ static void load(struct userdata *u) {
         char ln[256] = "";
         pa_sink *s;
 
-        (void) fgets(ln, sizeof(ln)-1, f);
-        pa_strip_nl(ln);
+        if (fgets(ln, sizeof(ln)-1, f))
+            pa_strip_nl(ln);
         fclose(f);
 
         if (!ln[0])
@@ -85,8 +85,8 @@ static void load(struct userdata *u) {
         char ln[256] = "";
         pa_source *s;
 
-        (void) fgets(ln, sizeof(ln)-1, f);
-        pa_strip_nl(ln);
+        if (fgets(ln, sizeof(ln)-1, f))
+            pa_strip_nl(ln);
         fclose(f);
 
         if (!ln[0])
diff --git a/src/pulsecore/authkey.c b/src/pulsecore/authkey.c
index 15613e2..f9cbd1e 100644
--- a/src/pulsecore/authkey.c
+++ b/src/pulsecore/authkey.c
@@ -56,7 +56,10 @@ static int generate(int fd, void *ret_data, size_t length) {
     pa_random(ret_data, length);
 
     lseek(fd, (off_t) 0, SEEK_SET);
-    (void) ftruncate(fd, (off_t) 0);
+    if (ftruncate(fd, (off_t) 0) < 0) {
+        pa_log("Failed to truncate cookie file: %s", pa_cstrerror(errno));
+        return -1;
+    }
 
     if ((r = pa_loop_write(fd, ret_data, length, NULL)) < 0 || (size_t) r != length) {
         pa_log("Failed to write cookie file: %s", pa_cstrerror(errno));
diff --git a/src/pulsecore/core-util.c b/src/pulsecore/core-util.c
index a642553..dab4e6c 100644
--- a/src/pulsecore/core-util.c
+++ b/src/pulsecore/core-util.c
@@ -243,11 +243,12 @@ int pa_make_secure_dir(const char* dir, mode_t m, uid_t uid, gid_t gid) {
     }
 
 #ifdef HAVE_FCHOWN
-    if (uid == (uid_t)-1)
+    if (uid == (uid_t) -1)
         uid = getuid();
-    if (gid == (gid_t)-1)
+    if (gid == (gid_t) -1)
         gid = getgid();
-    (void) fchown(fd, uid, gid);
+    if (fchown(fd, uid, gid) < 0)
+        goto fail;
 #endif
 
 #ifdef HAVE_FCHMOD
-- 
2.7.4

